go/cmd/dolt/commands/sql.go

```diff
@@ -240,6 +242,45 @@ func (cmd SqlCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 	return 0
 }
 
+func formatQueryError(query string, err error) errhand.VerboseError {
+	const (
+		maxStatementLen     = 128
+		maxPosWhenTruncated = 64
+	)
+
+	if se, ok := vterrors.AsSyntaxError(err); ok {
+		verrBuilder := errhand.BuildDError("Error parsing SQL")
+		verrBuilder.AddDetails(se.Message)
+
+		statement := se.Statement
+		position := se.Position
+		if len(se.Statement) > maxStatementLen {
+			if position > maxPosWhenTruncated {
+				statement = statement[position-maxPosWhenTruncated:]
+				position = maxPosWhenTruncated
+			}
+
+			if len(statement) > maxStatementLen {
+				statement = statement[:maxStatementLen]
+			}
+		}
+
+		verrBuilder.AddDetails(statement)
+
+		marker := make([]rune, position+1)
+		for i := 0; i < position; i++ {
+			marker[i] = ' '
+		}
+
+		marker[position] = '^'
+		verrBuilder.AddDetails(string(marker))
+
+		return verrBuilder.Build()
+	} else {
+		return errhand.BuildDError("Error Processing Query '%s'", query).AddCause(err).Build()
```

Comment:
Just capitalize the first word

Reply:
Actually, I think you just want to use errhand.VerboseErrorFromError here, rather than adding the underlying err as a cause. I don't think there's any good reason to be more verbose in the general case, let the client code decide what error message they want to print directly.

Reply:
updated